tests/test-filters.php

```diff
@@ -45,6 +45,70 @@ public function test_filter_database_queries() {
 		], WooCommerce_Custom_Orders_Table_Filters::filter_database_queries( $args, [] ) );
 	}
 
+	public function test_filter_database_queries_with_postmeta() {
+		$args = [
+			'meta_query' => [
+				'relation' => 'AND',
+				'customer_emails' => [
+					'key'     => '_billing_email',
+					'value'   => [ 'test@example.com' ],
+					'compare' => 'IN',
+				],
+				'custom_key' => [
+					'key' => '_custom_meta_key',
+					'value' => 'value',
+				],
+			],
+		];
+
+		$this->assertEquals( [
+			'meta_query' => $args['meta_query'],
+			'_wc_has_meta_columns' => true,
+			'wc_order_meta_query'  => [
+				[
+					'key'      => 'billing_email',
+					'value'    => [ 'test@example.com' ],
+					'compare'  => 'IN',
+					'_old_key' => '_billing_email',
+				],
+			],
+			'meta_query' => [
+				'relation' => 'AND',
+				'customer_emails' => [
+					'key'     => '_billing_email',
+					'value'   => [ 'test@example.com' ],
+					'compare' => 'IN',
+				],
+				'custom_key' => [
+					'key' => '_custom_meta_key',
+					'value' => 'value',
+				],
+			],
+		], WooCommerce_Custom_Orders_Table_Filters::filter_database_queries( $args, [] ) );
+	}
+
+	public function test_wc_get_orders_custom_meta_key() {
+
+		$order = WC_Helper_Order::create_order();
+		$order->update_meta_data( '_custom_meta_key', 'value' );
+		$order->save();
+
+		add_filter('woocommerce_order_data_store_cpt_get_orders_query', function ( $query, $query_vars ) {
```

Comment:
A bit of inline documentation might help to explain what's going on here — we're essentially saying "someone else has a callback on `woocommerce_order_data_store_cpt_get_orders_query` and we want to make sure we're not squashing anyone else's changes", correct?